### PR TITLE
[risk=low][no ticket] Split runtime-panel into components

### PR DESCRIPTION
**Pull request by jmthibault79**
**runtime-panel.tsx** was very large.  This PRs splits out most of its subcomponents.  The **new runtime-configuration-panel.tsx** is still very large!  but it's ~1000 instead of ~2500 lines, so hopefully that's helpful.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: ui/src/app/utils/runtime-utils.tsx [hunk 1]
```diff
@@ -1224,3 +1224,23 @@ export const withRuntimeStore = () => (WrappedComponent) => {
     return <WrappedComponent {...props} runtimeStore={value} />;
   };
 };
+
+export enum SparkConsolePath {
```
Comment (jmthibault79) [inline ui/src/app/utils/runtime-utils.tsx:1228]: these were also extracted from runtime-panel

File: ui/src/app/utils/runtime-utils.tsx [hunk 2]
```diff
@@ -1224,3 +1224,23 @@ export const withRuntimeStore = () => (WrappedComponent) => {
...
+  Yarn = 'yarn',
+  YarnTimeline = 'apphistory',
+  SparkHistory = 'sparkhistory',
+  JobHistory = 'jobhistory',
+}
+
+export enum PanelContent {
```
Comment (jmthibault79) [inline ui/src/app/utils/runtime-utils.tsx:1235]: extracted from runtime-panel

File: ui/src/app/pages/analysis/leonardo-app-launcher.tsx [hunk 3]
```diff
@@ -55,13 +56,6 @@ export enum LeoApplicationType {
   SparkConsole,
 }
 
-export enum SparkConsolePath {
```
Comment (jmthibault79) [inline ui/src/app/pages/analysis/leonardo-app-launcher.tsx:58]: moved to runtime-utils, on the principle that **components** should not depend on **pages**

File: ui/src/app/components/help-sidebar.tsx [hunk 4]
```diff
@@ -23,8 +23,8 @@ import { HelpTips } from 'app/components/help-tips';
 import { withErrorModal } from 'app/components/modals';
 import { TooltipTrigger } from 'app/components/popups';
 import { PopupTrigger } from 'app/components/popups';
+import { RuntimeConfigurationPanel } from 'app/components/runtime-configuration-panel';
```
Comment (jmthibault79) [inline ui/src/app/components/help-sidebar.tsx:26]: renamed the component

File: ui/src/app/components/runtime-configuration-panel/backup-files-help-section.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+
+import { supportUrls } from 'app/utils/zendesk';
+
+export const BackupFilesHelpSection = () => (
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel/backup-files-help-section.tsx:5]: This feels too small to need its own file, but 3 other components refer to it

File: ui/src/app/components/runtime-configuration-panel.tsx [hunk 6]
```diff
@@ -0,0 +1,1079 @@
+import * as React from 'react';
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.tsx:1]: moved ~2500 lines of `pages/analysis/runtime-panel` to:
* ~1000 lines to this file `components/runtime-configuration-panel.tsx`
* ~1500 lines to **components in the dir** `components/runtime-configuration-panel`

File: ui/src/app/components/runtime-configuration-panel.tsx [hunk 7]
```diff
@@ -0,0 +1,1079 @@
...
+        // selected value (it should always read "recommended environments"). The presets
+        // are not persistent state, they just snap the rest of the form to a particular configuration.
+        // See RW-5996 for more details.
+        return false;
+      }}
+    />
+  );
+};
+
+const PanelMain = fp.flow(
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.tsx:252]: this component is still 800 lines!  We may want to do another round of this...

**Comment by jmthibault79** [comment]
FYI @yonghaoy this may help make the runtime panel a little easier to deal with

**Review by yonghaoy — APPROVED** [review]

File: ui/src/app/components/runtime-configuration-panel.tsx (hunk 7 above)
Comment (jmthibault79) [reply ui/src/app/components/runtime-configuration-panel.tsx:252]: It took a year but we got there: #8032

**Pull request merged**